com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest.testIndexResolution

PR builder against master (commit b556d50e1a744b5b2c5f18b4b8d3d93a5fca89d3)

https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/9347/testReport/com.hazelcast.jet.sql.impl.connector.map.index/SqlIndexResolutionTest/testIndexResolution_indexType_SORTED__composite_true_/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<[QueryDataType {family=BOOLEAN}]> but was:<[]>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest.checkIndex(SqlIndexResolutionTest.java:211)
	at com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest.checkIndexResolution(SqlIndexResolutionTest.java:128)
	at com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest.testIndexResolution(SqlIndexResolutionTest.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
13:40:23,632  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
13:40:23,634  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
13:40:23,634  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
13:40:23,634  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Hazelcast Platform 5.1-SNAPSHOT (20220121 - 83283a6) starting at [127.0.0.1]:5701
13:40:23,634  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Cluster name: dev
13:40:23,641  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
13:40:23,649  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
13:40:23,653  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
13:40:23,656  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
13:40:23,656  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
13:40:23,657  INFO || - [JetExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Jet is enabled
13:40:23,657  INFO || - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353 this
]

13:40:23,657 DEBUG || - [JobCoordinationService] hz.wizardly_lewin.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
13:40:23,657  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
13:40:23,658  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
13:40:23,659  INFO || - [logo] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
13:40:23,659  INFO || - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
13:40:23,659  INFO || - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Hazelcast Platform 5.1-SNAPSHOT (20220121 - 83283a6) starting at [127.0.0.1]:5702
13:40:23,659  INFO || - [system] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Cluster name: dev
13:40:23,660  INFO || - [HealthMonitor] hz.wizardly_lewin.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=16, physical.memory.total=377.6G, physical.memory.free=32.0G, swap.space.total=4.0G, swap.space.free=3.9G, heap.memory.used=300.9M, heap.memory.free=552.1M, heap.memory.total=853.0M, heap.memory.max=1.8G, heap.memory.used/total=35.22%, heap.memory.used/max=16.50%, minor.gc.count=13, minor.gc.time=312ms, major.gc.count=3, major.gc.time=553ms, load.process=6.67%, load.system=100.00%, load.systemAverage=57.00, thread.count=36, thread.peakCount=62, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=0, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=1, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
13:40:23,664  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
13:40:23,670  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
13:40:23,675  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
13:40:23,677  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
13:40:23,678  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5702 is STARTING
13:40:23,678  INFO || - [MockServer] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701-08d467b0-1f0a-45e8-8a4b-0efb9329b353, connection: MockConnection{localEndpoint=[address=[127.0.0.1]:5702, uuid=aefb6461-006c-4bf4-ab0f-774c04a6cdc6], remoteEndpoint=[address=[127.0.0.1]:5701, uuid=08d467b0-1f0a-45e8-8a4b-0efb9329b353], alive=true}
13:40:23,680  INFO || - [MockServer] hz.wizardly_lewin.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702-aefb6461-006c-4bf4-ab0f-774c04a6cdc6, connection: MockConnection{localEndpoint=[address=[127.0.0.1]:5701, uuid=08d467b0-1f0a-45e8-8a4b-0efb9329b353], remoteEndpoint=[address=[127.0.0.1]:5702, uuid=aefb6461-006c-4bf4-ab0f-774c04a6cdc6], alive=true}
13:40:23,681  INFO || - [ClusterService] hz.wizardly_lewin.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353 this
	Member [127.0.0.1]:5702 - aefb6461-006c-4bf4-ab0f-774c04a6cdc6
]

13:40:23,757 DEBUG || - [JobCoordinationService] hz.wizardly_lewin.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
13:40:23,780  INFO || - [JetExtension] hz.vigorous_lewin.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] Jet is enabled
13:40:23,782  INFO || - [ClusterService] hz.vigorous_lewin.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353
	Member [127.0.0.1]:5702 - aefb6461-006c-4bf4-ab0f-774c04a6cdc6 this
]

13:40:23,783  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5702 is STARTED
Started Running Test: testIndexResolution[indexType:SORTED, composite:true]
13:40:23,785  INFO || - [HealthMonitor] hz.vigorous_lewin.HealthMonitor - [127.0.0.1]:5702 [dev] [5.1-SNAPSHOT] processors=16, physical.memory.total=377.6G, physical.memory.free=32.0G, swap.space.total=4.0G, swap.space.free=3.9G, heap.memory.used=310.7M, heap.memory.free=542.3M, heap.memory.total=853.0M, heap.memory.max=1.8G, heap.memory.used/total=36.36%, heap.memory.used/max=17.04%, minor.gc.count=13, minor.gc.time=312ms, major.gc.count=3, major.gc.time=553ms, load.process=7.14%, load.system=100.00%, load.systemAverage=57.00, thread.count=62, thread.peakCount=62, cluster.timeDiff=-3, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=13, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=1, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
13:40:23,827  INFO |testIndexResolution[indexType:SORTED, composite:true]| - [PartitionStateManager] testIndexResolution - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Initializing cluster partition table arrangement...
13:40:23,861 DEBUG || - [JobRepository] hz.wizardly_lewin.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
13:40:24,066 ERROR || - [EventServiceImpl] hz.wizardly_lewin.event-9 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] hz.wizardly_lewin.event-9 caught an exception while processing:com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher@58259065
java.lang.UnsupportedOperationException: SQL catalog must never be cleared - MapEvent{entryEventType=CLEAR_ALL, member=Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353 this, name='__sql.catalog', numberOfEntriesAffected=2}
	at com.hazelcast.jet.sql.impl.schema.TablesStorage$EntryListenerAdapter.mapCleared(TablesStorage.java:202) ~[classes/:?]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapEventPublishingService.dispatchEvent(ReplicatedMapEventPublishingService.java:132) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapService.dispatchEvent(ReplicatedMapService.java:232) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher.run(LocalEventDispatcher.java:64) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:245) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:228) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
13:40:24,066 ERROR || - [EventServiceImpl] hz.wizardly_lewin.event-9 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] hz.wizardly_lewin.event-9 caught an exception while processing:com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher@56e26511
java.lang.UnsupportedOperationException: SQL catalog must never be cleared - MapEvent{entryEventType=CLEAR_ALL, member=Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353 this, name='__sql.catalog', numberOfEntriesAffected=2}
	at com.hazelcast.jet.sql.impl.schema.TablesStorage$EntryListenerAdapter.mapCleared(TablesStorage.java:202) ~[classes/:?]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapEventPublishingService.dispatchEvent(ReplicatedMapEventPublishingService.java:132) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapService.dispatchEvent(ReplicatedMapService.java:232) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher.run(LocalEventDispatcher.java:64) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:245) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:228) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
13:40:24,067 ERROR || - [EventServiceImpl] hz.wizardly_lewin.event-9 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] hz.wizardly_lewin.event-9 caught an exception while processing:com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher@52ddfb15
java.lang.UnsupportedOperationException: SQL catalog must never be cleared - MapEvent{entryEventType=CLEAR_ALL, member=Member [127.0.0.1]:5701 - 08d467b0-1f0a-45e8-8a4b-0efb9329b353 this, name='__sql.catalog', numberOfEntriesAffected=2}
	at com.hazelcast.jet.sql.impl.schema.TablesStorage$EntryListenerAdapter.mapCleared(TablesStorage.java:202) ~[classes/:?]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapEventPublishingService.dispatchEvent(ReplicatedMapEventPublishingService.java:132) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapService.dispatchEvent(ReplicatedMapService.java:232) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher.run(LocalEventDispatcher.java:64) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:245) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:228) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
BuildInfo right after testIndexResolution[indexType:SORTED, composite:true](com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest): BuildInfo{version='5.1-SNAPSHOT', build='20220121', buildNumber=20220121, revision=83283a6, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testIndexResolution[indexType:SORTED, composite:true](com.hazelcast.jet.sql.impl.connector.map.index.SqlIndexResolutionTest):'
13:40:20, accumulated pauses: 373 ms, max pause: 93 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/index/SqlIndexResolutionTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/index/SqlIndexResolutionTest.java
@@ -124,7 +124,7 @@ public class SqlIndexResolutionTest extends SqlIndexTestSupport {
         map = nextMap();
         value = ExpressionBiValue.createBiValue(valueClass, 1, f1.valueFrom(), null);
         createMapping(map.getName(), int.class, value.getClass());
-        map.put(1, value);
+        putValues(map, value);
         checkIndex(map, f1.getFieldConverterType());
         checkIndexUsage(map, f1, f2, true, false);
         map.destroy();
@@ -134,7 +134,7 @@ public class SqlIndexResolutionTest extends SqlIndexTestSupport {
             map = nextMap();
             value = ExpressionBiValue.createBiValue(valueClass, 1, null, f2.valueFrom());
             createMapping(map.getName(), int.class, value.getClass());
-            map.put(1, value);
+            putValues(map, value);
             checkIndex(map);
             checkIndexUsage(map, f1, f2, false, true);
             map.destroy();
@@ -143,10 +143,17 @@ public class SqlIndexResolutionTest extends SqlIndexTestSupport {
             map = nextMap();
             value = ExpressionBiValue.createBiValue(valueClass, 1, f1.valueFrom(), f2.valueFrom());
             createMapping(map.getName(), int.class, value.getClass());
-            map.put(1, value);
+            putValues(map, value);
             checkIndex(map, f1.getFieldConverterType(), f2.getFieldConverterType());
             checkIndexUsage(map, f1, f2, true, true);
             map.destroy();
+        }
+    }
+
+    private void putValues(IMap<Integer, ExpressionBiValue> map, ExpressionBiValue value) {
+        // have enough entries to make sure that each member has at least one.
+        for (int i = 0; i < 100; i++) {
+            map.put(i, value);
         }
     }
 

